pyalgostrategypool/strategy_options_bull_call_spread_with_target_stops_reentry/_strategy.py

```diff
@@ -0,0 +1,229 @@
+    """
+    Bull Call Spread strategy with multiple exit mechanisms:
+        • Target Profit
+        • Hard Stop-Loss
+        • Trailing Stop-Loss
+        • Optional Re-entry
+    """
+
+    name = "Strategy Options Bull Call Spread With Target Stops And Re-entry"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Required parameters for this strategy
+        self.no_of_otm_strikes_leg_sell = self.strategy_parameters["NUMBER_OF_OTM_STRIKES_SELL_LEG"]
+        self.stoploss_percentage = self.strategy_parameters['STOPLOSS_PERCENTAGE']
+        self.target_percentage = self.strategy_parameters['TARGET_PERCENTAGE']
+        self.tsl_percentage = self.strategy_parameters["TRAILING_STOPLOSS_PERCENTAGE"]
+        self.re_entry_limit = self.strategy_parameters['RE_ENTRY_LIMIT']
+
+        # Internal variables and placeholders
+        self.child_instrument_main_orders = None  # Tracks Call orders
+        self.number_of_allowed_expiry_dates = 1  # Restrict how many expiry dates can be used
+        self.validate_parameters()
+
+    def validate_parameters(self):
+        """ Validates required strategy parameters. """
+        check_argument(
+            self.strategy_parameters, "extern_function", lambda x: len(x) >= 2,
+            err_message="Need 5 parameters for this strategy: \n(1) NUMBER_OF_OTM_STRIKES_SELL_LEG \n(2) TARGET_PERCENTAGE \n(3) STOPLOSS_PERCENTAGE \n(4) TRAILING_STOPLOSS_PERCENTAGE \n(5) RE_ENTRY_LIMIT"
+        )
+
+        # Validate expiry dates
+        if len(self.get_allowed_expiry_dates()) != self.number_of_allowed_expiry_dates:
+            self.logger.info(f"Allowed expiry dates: {self.number_of_allowed_expiry_dates}, got {len(self.get_allowed_expiry_dates())}. Exiting...")
+            raise ABSystemExit
+
+        # Validate parameters
+        for param in (self.re_entry_limit, self.tsl_percentage):
+            check_argument(param, "extern_function", is_nonnegative_int_or_float, "Value should be >0")
+
+    def initialize(self):
+        super().initialize()
+
+        # Reset main orders, trailing stops and re-entry counts
+        self.child_instrument_main_orders = {}
+        self.spread_current = self.spread_entry = self.highest = self.trailing_stop = self.stoploss_premium = self.target_premium = None
+        self.re_entry_count = {}
+
+    def check_exit_conditions(self, base_instrument):
+        """
+        Evaluate all exit rules for the Bull Call Spread.
+
+        Checks:
+        • Target profit – exit if spread rises to the profit target.
+        • Hard stop-loss – exit if spread falls below the stop-loss threshold.
+        • Trailing stop-loss – once the spread makes new highs, trail a stop to lock in profits.
+        """
+
+        ltp_leg_buy = self.broker.get_ltp(self.child_instrument_main_orders.get(base_instrument)[BrokerOrderTransactionTypeConstants.BUY].instrument)
```

Comment:
You can define this as a variable and use it on 77 and 82 to keep the lines shorter and simpler to read:

`self.child_instrument_main_orders.get(base_instrument)[BrokerOrderTransactionTypeConstants.BUY]`